src/metabase/api/collection.clj

```diff
@@ -187,18 +207,40 @@
   [id]
   (collection-detail (api/read-check Collection id)))
 
+(defn- coerce-vec [to-coerce]
+  (if (and (some? to-coerce) (not (vector? to-coerce))) [to-coerce] to-coerce))
```

Comment:
We do this frequently enough that we a have a function in `metabase.util` to do this. `u/one-or-many` does the same thing.

Either way, I'd recommend checking `sequential?` rather than `vector?`... it's better to be flexible and handle any kind of sequence e.g. a lazy sequence or list rather than only vectors